File: java_solutions/2500.delete-greatest-value-in-each-row.java
/*
 * @lc app=leetcode id=2500 lang=java
 *
 * [2500] Delete Greatest Value in Each Row
 */

class Solution {
    public int deleteGreatestValue(int[][] grid) {
        return IntStream.range(0, grid[0].length)
            .map(j -> IntStream.range(0, grid.length)
                .map(i -> Arrays.stream(grid[i]).sorted().toArray()[j])
                .max().getAsInt())
            .sum();
    }
}

File: kotlin_solutions/2500.delete-greatest-value-in-each-row.kt
/*
 * @lc app=leetcode id=2500 lang=kotlin
 *
 * [2500] Delete Greatest Value in Each Row
 */

class Solution {fun deleteGreatestValue(grid: Array<IntArray>) = grid.onEach { it.sort() }.let { g -> (0 until g[0].size).sumOf { j -> g.maxOf { it[j] } } }}

